scan/context.go

```diff
@@ -113,6 +120,29 @@ func (s *Scanner) getContextFromReader(r io.Reader) (err error) {
 	return err
 }
 
+func (s *Scanner) getContextFromRegistry(ctx context.Context, registryArtifact string) (err error) {
+	resolver := docker.NewResolver(docker.ResolverOptions{
+		PlainHTTP: false,
+		Client:    http.DefaultClient,
+		Credentials: func(hostName string) (string, string, error) {
```

Comment:
Suggestion: docker.Credentials is deprecated. Can you try using RegistryHosts..

I am not sure if it works, but we can try
```go
authorizer  := docker.NewAuthorizer(http.DefaultClient, func(hostName string) (string, string, error) {
    // If no matching credential found, attempt an anonymous pull
	if s.credentials[hostName] == nil {
		return "", "", nil
	}
	return s.credentials[hostName].Username.ResolvedValue, s.credentials[hostName].Password.ResolvedValue, nil
})
resolver := docker.NewResolver(docker.ResolverOptions{
        PlainHTTP: false,
	Hosts: docker.ConfigureDefaultRegistries(
		docker.WithAuthorizer(authorizer),
	),
})
```

Reply:
Switched to this and works locally.